### DISCUSSION TITLE
Deleting legends fails

### DISCUSSION THREAD

**Discussion by bilderbuchi** [JuliaPlots/Makie.jl#2657] [Q&A] [answered]
Having the same problem as reported [here](https://discourse.julialang.org/t/makie-how-to-remove-a-legend-from-an-axis/65876), i.e. wanting to update plotted lines including the legend, I ran into the issue that a `delete!(legend)` does not work.
With the code from that thread
```julia
using GLMakie
f = Figure()
ax1 = f[1,1] = Axis(f)
ax2 = f[1,2] = Axis(f)
p1 = lines!(ax1, rand(10), label = "foo")
p2 = lines!(ax2, rand(10), label = "bar")
function add_a_legend!(ax) axislegend(ax); return nothing end
add_a_legend!(ax1)
add_a_legend!(ax2)
empty!(ax1)
[delete!(leg) for leg in f.content if leg isa Legend]
```
I get
```
ERROR: MethodError: no method matching remove_from_gridlayout!(::Nothing)
Closest candidates are:
  remove_from_gridlayout!(::GridLayoutBase.GridContent) at C:\<snip>\.julia\packages\GridLayoutBase\lYdxT\src\gridlayout.jl:290
Stacktrace:
 [1] delete!(block::Legend)
   @ Makie C:\Users\<snip>\.julia\packages\Makie\gAmAB\src\makielayout\blocks.jl:508
 [2] (::var"#13#15")(leg::Legend)
   @ Main .\none:0
 [3] iterate
   @ .\generator.jl:47 [inlined]
 [4] grow_to!(dest::Vector{Any}, itr::Base.Generator{Base.Iterators.Filter{var"#14#16", Vector{Any}}, var"#13#15"})
   @ Base .\array.jl:860
 [5] collect(itr::Base.Generator{Base.Iterators.Filter{var"#14#16", Vector{Any}}, var"#13#15"})
   @ Base .\array.jl:784
 [6] top-level scope
   @ Untitled-1:11
```
from https://github.com/MakieOrg/Makie.jl/blob/b1b9ef2923bea6a2b9c311cb2d17dbdb49674003/src/makielayout/blocks.jl#L508
This is with Makie 0.19.2, Julia 1.8.5.

Am I doing something wrong, or is this a bug?

**Comment by jkrumbiegel** [comment, marked as answer]
Probably just a bug, because deletion is currently untested. It would be a good project to start and make these methods more correct and robust, and fix that behavior via tests so that it doesn't break at the next moment again. This hasn't been the priority historically, because we had much more to do with _creating_ things before we wanted to _delete_ them :)

**Reply by bilderbuchi** [reply]
Thanks, I'll report it then.
edit: #2659
